File: isSubsequence/isSubsequence.go
package main

import (
	"fmt"
)

func isSubsequence(s, t string) bool {
	var cnt int

	if len(s) == 0 {
		return true
	}

	if len(t) < len(s) {
		return false
	}

	for i := 0; i < len(t); i++ {
		if t[i] == s[cnt] {
			cnt++
		}
	}

	if cnt < len(s) {
		return false
	}

	return true
}

func main() {
	fmt.Println(isSubsequence("abc", "ahbgdc"))
	fmt.Println(isSubsequence("axc", "ahbgdc"))
}

File: isSubsequence/isSubsequence.cpp
#include <iostream>

using namespace std;

class Solution {
public:
    bool isSubsequence(string s, string t) {
        for (int i = 0, j = 0; i < s.size(); i++) {
            while (j < t.size() && s[i] != t[j]) j++;
            if (j == t.size()) return false;
            j++;
        }
        return true;
    }
};

int main() {
    string s = "abc";
    string t = "ahbgdc";
    Solution sol;
    cout << sol.isSubsequence(s, t) << endl;
    return 0;
}